synapse/events/snapshot.py

```diff
@@ -106,21 +106,15 @@ class EventContext(UnpersistedEventContextBase):
         state_delta_due_to_event: If `state_group` and `state_group_before_event` are not None
             then this is the delta of the state between the two groups.
 
-        prev_group: If it is known, ``state_group``'s prev_group. Note that this being
-            None does not necessarily mean that ``state_group`` does not have
-            a prev_group!
-
-            If the event is a state event, this is normally the same as
-            ``state_group_before_event``.
-
-            If ``state_group`` is None (ie, the event is an outlier), ``prev_group``
-            will always also be ``None``.
-
-            Note that this *not* (necessarily) the state group associated with
-            ``_prev_state_ids``.
-
-        delta_ids: If ``prev_group`` is not None, the state delta between ``prev_group``
-            and ``state_group``.
+        state_group_deltas: if it exists, this is a dict collecting a mapping of two possible
+            sets of state groups and their deltas: the state group before the event and
+            after the event and the state map that represents the state group between the
+            two, i.e. (state_group_2, state_group_1) -> (state_key, event_type) -> event_id
+            or/and the state group preceding the state group before the event and the state
+            group before the event and the state map that represents the delta between those
+            two state groups.
+            This information is collected and stored as part of an optimization for persisting
+            events.
```

Comment:
I'm still finding this really hard to understand, I think it needs some whitespace and less complex sentences. I've rewritten it, but I don't know if it is accurate.

```suggestion
        state_group_deltas: if it exists, this is a dict collecting a mapping of the state
            difference between state groups.

            The keys are a tuple of two integers: the initial group and final state group.
            The corresponding value is a state map representing the state delta between
            these state groups.

            The dictionary is expected to have at most two entries with state groups of:

            1. The state group before the event and after the event.
            2. The state group preceding the state group before the event and the
               state group before the event.

            This information is collected and stored as part of an optimization for persisting
            events.
```

Reply:
Much clearer, thank you!